beacon-chain/cache/depositsnapshot/deposit_tree.go

```diff
@@ -119,9 +117,6 @@ func (d *DepositTree) getProof(index uint64) ([32]byte, [][32]byte, error) {
 		return [32]byte{}, nil, ErrInvalidMixInLength
 	}
 	finalizedDeposits, _ := d.tree.GetFinalized([][32]byte{})
-	if finalizedDeposits == 0 {
-		return [32]byte{}, nil, ErrNoFinalizedDeposits
-	}
```

Comment:
Why remove this? Does having this piece of code fail a spec test?

Reply:
I removed it because having 0 finalized deposits is valid. And we wont have any issues because we have a check right after it and before we try to subtract `finalizedDeposits` by 1.
```
if finalizedDeposits != 0 {
		finalizedDeposits = finalizedDeposits - 1
	}
```